contrib/inventory/puppetdb.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+#
+# Ansible inventory script based on puppetdb.
+#
+# Author: Jan Collijs
+# Source: https://github.com/visibilityspots/ansible-puppet-inventory
+
+import os
+import argparse
+from pypuppetdb import connect
+
+try:
+    import ConfigParser
+except ImportError:
+    import configparser as ConfigParser
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+
+class PuppetDBInventory:
+    """This class will generate a list of nodes based on puppetdb"""
+    def __init__(self):
+        self.read_settings()
+        self.parse_cli_args()
+        if self.args.list:
```

Comment:
I noticed this earlier, but forgot to comment on it.

In the event someone wants to import your inventory script for re-use or unit testing, it would be better to have the argument parsing, inventory retrieval and printing called from `main` instead of `__init__`.

So something like this in `PuppetDBInventory`:

```python
def print_inventory(self):
    data = self.get_host_list_based_on_environments()
    print(json.dumps(data))
```

Then move the argument handling logic out of `PuppetDBInventory`:

```python
def parse_cli_args():
    """Parsing the parameters given through the command line"""
    parser = argparse.ArgumentParser(description='Produce an Ansible Inventory file based on puppetdb')
    parser.add_argument('--list', action='store_true', default=True,
                        help='List instances (default: True)')
    return parser.parse_args()

def main():
    """Main class which initiates the whole script"""
    args = parse_cli_args()
    if args.list:
        inventory = PuppetDBInventory()
        inventory.print_inventory()  
```

I realize the way it's currently implemented matches most of the other inventory scripts. However, it appears that someone just started with that pattern and everyone else has been copying it.